ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -1205,6 +1205,10 @@ Are you sure you want to disable auto-sync and rollback application '${this.prop
                 if (needDisableRollback) {
                     const update = JSON.parse(JSON.stringify(application)) as appModels.Application;
                     update.spec.syncPolicy = {automated: null};
+                    // preserve syncOptions when doing rollback
+                    if (application.spec.syncPolicy.syncOptions) {
+                        update.spec.syncPolicy.syncOptions = application.spec.syncPolicy.syncOptions;
+                    }
```

Comment:
Would this work? This way the same problem does not happen if new properties are added to syncPolicy.

```suggestion
                    update.spec.syncPolicy = application.spec.syncPolicy;
                    update.spec.syncPolicy.automated = null;
```

Reply:
```suggestion
                   set('spec.syncPolicy.syncOptions.automated', null)
```
It seems we already clone syncPolicy from the application, so just setting automated to null should be enough instead of overriding the whole syncPolicy.🤔